### ISSUE TITLE
Document how bad --no-tls-very-insecure is

### ISSUE THREAD

**Issue by AArnott** [zcash/lightwalletd#423]
The `--no-tls-very-insecure` is duly self-described as "very insecure", but it doesn't explain why. It only deals with publicly available information. No keys are ever sent to or from it. So why is it so insecure to use it without TLS?

In particular, I've set up my litewalletd server with an nginx front-end that adds HTTPS as a reverse proxy. So my plan is in fact to use `--no-tls-very-insecure` in production, because I believe there is no need to encrypt litewalletd traffic between itself and my nginx front-end that is on the same network, given nginx itself will encrypt everything to each client.

Can you confirm this is a reasonable approach?

**Comment by LarryRuane** [comment]
Yes, that is reasonable.  The option was named in that way just so `lightwalletd` node operators would understand that in this mode, traffic between the `lightwalletd` and its clients would not be encrypted with TLS.  (I actually run in this mode all the time within my own machine while testing, so I can confirm that it works.)

I'll close this now but feel free to comment and reopen if you have any further questions or concerns.

**Comment by defuse** [comment]
@AArnott your approach sounds reasonable to me. 

The biggest reasons why a wallet connecting to a lightwalletd over an unauthenticated and unencrypted connection, even though the blockchain is all public information, are (a) most wallets currently do not authenticate the block/transaction data coming from lightwalletd, so a MitM could send "fake money" to the wallets, and (b) most wallets currently reveal which shielded transactions belong to them when fetching the memos, so a MitM could learn which transactions belong to which wallets.

In other words, the lightwalletd server is partially trusted to maintain privacy and data authenticity, so using an unencrypted connection would put wallets at risk. There's more detailed information in the threat model: https://zcash.readthedocs.io/en/latest/rtd_pages/wallet_threat_model.html
